Add option to ignore CSS files starting with _

florianeckerstorfer opened the pull request:
Hey, first thanks for the great plugin. Works, like a charm.

In my setup I use `postcss-import` and I use the convention to start imported files with an underscore (which I think is quite common). To keep my final build clean, I don't want these included files out of my final build, so I added an option to the plugin to ignore files that start with an underscore.

You would use the option like this:
```js
// Import the plugin
const PostCSSPlugin = require("eleventy-plugin-postcss");

module.exports = (config) => {
    // Enable the plugin in you project
    config.addPlugin(PostCSSPlugin, { ignoreUnderscore: true });
}
```

I had to make a change to how the extension is being called internally, so that I can pass the `pluginOptions` to the extension.

I think this would also close #7

whoisvadym commented:
Hi @florianeckerstorfer,
Thanks for your feedback and your Pull Request. I'll review your changes and will get back to you asap.

florianeckerstorfer commented:
@whoisvadym Let me know if there are any tests I should add

whoisvadym commented on README.md:
```diff
@@ -27,5 +27,16 @@ module.exports = (config) => {
 ## Configuration
 To load and resolve the configuration file the plugin uses [`postcss-load-config`](https://github.com/postcss/postcss-load-config) module. Check out the project [readme](https://github.com/postcss/postcss-load-config#readme) for the complete list of supported config file formats.
 
+If you want to ignore CSS files that start with `_` to be processed, you can pass the `ignoreUnderscore` option to the plugin:
```
Thanks for updating the readme!

whoisvadym commented on src/extension.js:
```diff
@@ -1,18 +1,25 @@
     return async ({ page }) => {
+      if (
+        pluginOptions.ignoreUnderscore 
+        && path.basename(page.inputPath).match(/^_/)
```
I think it would be more flexible to let developers provide the pattern for matching in case someone has a different naming convention for files that they would like to exclude. So something like `pluginOptions.ignorePattern` or `pluginOptions.ignoreRegex`. What do you think?

whoisvadym commented on src/extension.js:
```diff
@@ -1,18 +1,25 @@
   async compile(input, inputPath) {
     return async ({ page }) => {
+      if (
+        pluginOptions.ignoreUnderscore 
```
Please address the pipeline failure (`Trailing spaces not allowed`)

whoisvadym commented on .eleventy.js:
```diff
@@ -1,9 +1,12 @@
 // eslint-disable-next-line no-unused-vars, import/no-extraneous-dependencies
 const EleventyConfig = require('@11ty/eleventy/src/UserConfig');
-const Extension = require('./src/extension');
+const extension = require('./src/extension');
```
👍

whoisvadym commented on README.md:
```diff
@@ -27,5 +27,16 @@ module.exports = (config) => {
+
+module.exports = (config) => {
+    // Enable the plugin in you project
+    config.addPlugin(PostCSSPlugin, { ignoreUnderscore: true });
```
Please add the use case for the new ignore functionality in the integration tests.

whoisvadym commented on src/extension.js:
```diff
@@ -1,18 +1,25 @@
 const OUTPUT_FILE_EXTENSION = 'css';
 
-module.exports = {
+module.exports = (pluginOptions = {}) => ({
```
This should also fall back to a `DEFAULT_PLUGIN_OPTIONS`.

whoisvadym reviewed (changes requested):
Hi @florianeckerstorfer, thanks for taking the time to contribute to this project. I approved the pipeline, and it looks like one formatting issue needs a resolution. Please take a look at a code review for more details.

florianeckerstorfer commented:
Hey @whoisvadym, thanks for the feedback. I fixed the formatting issues and renamed the option to `ignorePattern`. I also added a test case to the integration test, but based on how the tests are currently set up, it is tricky to check different values for the option

whoisvadym commented:
After considering this further, I'm going to close this PR. For the sake of simplicity, I'd like to avoid bringing any sort customization into this plugin unless absolutely necessary. 

In most basic scenarios excluding certain files can be as simple as slightly changing the project structure. ([Example](https://github.com/whoisvadym/eleventy-plugin-postcss/issues/7#issuecomment-2610620498))

Please feel free to [Open a new Issue](https://github.com/whoisvadym/eleventy-plugin-postcss/issues/new/) if you think your use-case requires ignoring the files at plugin level.
